cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -244,6 +244,22 @@ class aggregate_reader_metadata {
     host_span<std::vector<size_type> const> row_group_indices,
     host_span<int const> column_schemas) const;
 
+  /**
+   * @brief Filters the row groups using row bounds (`skip_rows` and `num_rows`)
+   *
+   * @param rows_to_skip Number of rows to skip
+   * @param rows_to_read Number of rows to read
+   *
+   * @return A tuple of surviving row group indices, and two vectors of effective (trimmed) row
+   * counts and offsets across surviving row group indices respectively
```

Comment:
How do we have an element per row group in the whole input? Don't we only need to skip rows in the first selected row group, and limit the number of rows in the last selected row group?

Reply:
Yes, but we might end up removing more row groups from the termini in the predicate pushdown stage so we need the whole range so that we know where the first row group starts once we are done filtering for real.